src/core/index.tsx

```diff
@@ -31,15 +31,27 @@ class Core {
       this.lightboxOpacity = opts.lightboxOpacity || 0.4;
       this.providerOptions = opts.providerOptions || {};
     }
-
+    
     this.renderModal();
   }
 
-  public on(event: string, callback: (result: any) => void): void {
+  public on(event: string, callback: (result: any) => void): () => void {
     this.eventManager.on({
       event,
       callback
     });
+
+    return () => this.eventManager.off({
```

Comment:
Why do you return a callback to remove it? Just curious

Reply:
I think it's a good practice to make `unsubscribe` function readily available. For example, for easier cleanup in a `useEffect` hook:
```
const core = new Web3Connect.Core({
...
useEffect(() => {
  return core.on('connect', provider => setWeb3(new Web3(provider)))
}, [])
```
No need to keep a reference to ` provider => setWeb3(new Web3(provider))` to pass to `core.off('connect',...)`